plugins/kubernetes/app/models/kubernetes/deploy_executor.rb

```diff
@@ -154,9 +142,9 @@ def show_logs_on_deploy_if_requested(statuses)
     def show_failure_cause(release_docs, statuses)
       release_docs.each { |doc| print_resource_events(doc) }
       log_end_time = Integer(ENV['KUBERNETES_LOG_TIMEOUT'] || '20').seconds.from_now
-      debug_pods = statuses.reject(&:live).select(&:pod).group_by(&:role).map { |_, g| g.first.pod }
-      debug_pods.each do |pod|
-        print_pod_details(pod, log_end_time, events: true)
+      debug_statuses = statuses.reject(&:live).select(&:resource).group_by(&:role).each_value.map(&:first)
+      debug_statuses.each do |status|
+        print_pod_details(status, log_end_time, events: true)
```

Comment:
are these resources guaranteed to be pods?

Reply:
yes